linux/net/rina/dtcp-utils.c

```diff
@@ -346,6 +349,14 @@ static struct dtcp_config * dtcp_config_create_gfp(gfp_t flags)
           }
         */
 
+        tmp->dtcp_ps = policy_create();
```

Comment:
Yes, actually there is a FIXME for that:

 645 /* FIXME This function has not been ported yet to use the DTCP policy set
 646  * parameters in place of struct dtcp_config. This because the code
 647  * tries to access "connection parameters" that are defined as DTCP
 648  * policies (in RINA) even if DTCP is not present. This has to
 649  * be fixed, because the DTCP policy set can exist only if DTCP
 650  * is present.
 651  */
 
we have to check if all the data that is later use will be present somewhere else (ps, state vector, etc). If his is the case, we can free the config structs once the corresponding instances are created and the config parameters that are later neede are passed somewhere. As you said, this is another issue, which could be created to address later on.